src/kademlia/dht_tracker.cpp

```diff
@@ -585,40 +557,31 @@ namespace libtorrent { namespace dht
 
 	void add_node_fun(void* userdata, node_entry const& e)
 	{
-		entry* n = static_cast<entry*>(userdata);
-		std::string node;
-		std::back_insert_iterator<std::string> out(node);
-		write_endpoint(e.ep(), out);
-		n->list().push_back(entry(node));
+		auto v = static_cast<std::vector<udp::endpoint>*>(userdata);
+		v->push_back(e.ep());
 	}
 
-	void save_nodes(entry& ret, node const& dht, std::string const& key)
+	void save_nodes(std::vector<udp::endpoint>& ret, node const& dht)
```

Comment:
you wouldn't even need to use ``std::move()``, since it would be an implied rvalue reference in the ``return`` statement.